src/runtime/virtcontainers/pkg/cgroups/utils.go

```diff
@@ -75,6 +80,76 @@ func IsSystemdCgroup(cgroupPath string) bool {
 	return false
 }
 
+func newProperty(name string, units interface{}) systemdDbus.Property {
+	return systemdDbus.Property{
+		Name:  name,
+		Value: dbus.MakeVariant(units),
+	}
+}
+
+func cgroupHierarchy(path string) (cgroups.Hierarchy, cgroups.Path, error) {
+	if !IsSystemdCgroup(path) {
+		return cgroups.V1, cgroups.StaticPath(path), nil
+	} else {
+		slice, unit, err := getSliceAndUnit(path)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		cgroupSlicePath, _ := systemd.ExpandSlice(slice)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		return cgroups.Systemd, cgroups.Slice(cgroupSlicePath, unit), nil
+	}
+}
+
+func createCgroupsSystemd(slice string, unit string, pid uint32) error {
+	ctx := context.TODO()
+	conn, err := systemdDbus.NewWithContext(ctx)
+	if err != nil {
+		return err
+	}
+	defer conn.Close()
+
+	properties := []systemdDbus.Property{
+		systemdDbus.PropDescription("cgroup " + unit),
+		newProperty("DefaultDependencies", false),
+		newProperty("MemoryAccounting", true),
+		newProperty("CPUAccounting", true),
+		newProperty("BlockIOAccounting", true),
```

Comment:
You may want to use `IOAccounting` now (see https://www.freedesktop.org/software/systemd/man/systemd.resource-control.html). I think `BlockIOAccounting` is deprecated.

Do we want to enable `IPAccounting` as well?

Reply:
I'm not sure, in my environment network metrics are currently being collected (from another source i guess), if someone uses it i can enable it